code/modules/projectiles/bullet.dm

```diff
@@ -816,19 +817,31 @@ toxic - poisons
 	implanted = null
 	damage = 6
 
+	New()
+		..()
+		if(prob(50)) //let's get some variation in here
+			icon_state = "glass2"
+		if(prob(20)) //we want implants, but we only want a few
+			implanted = /obj/item/implant/projectile/glass_shard
```

Comment:
This is not how this works, `/datum/projectile`s are stored on the gun while `/obj/projectile`s are initialized per shot. If you want to randomize the icon state you can do that in `on_launch` and if you want to have `prob` based implants you'll have to implement that in the projectile implanting code.